--- a/instrumentation/nginx/CMakeLists.txt
+++ b/instrumentation/nginx/CMakeLists.txt
@@ -1,6 +1,8 @@
+set(NGINX_VERSION $ENV{NGINX_VERSION})
 cmake_minimum_required(VERSION 3.12)

 project(opentelemetry-nginx)
+include(${CMAKE_BINARY_DIR}/conan_paths.cmake)

 find_package(opentelemetry-cpp REQUIRED)
 find_package(Threads REQUIRED)
diff --git a/instrumentation/nginx/nginx.cmake b/instrumentation/nginx/nginx.cmake
index db1a168..ec8e84f 100644
--- a/instrumentation/nginx/nginx.cmake
+++ b/instrumentation/nginx/nginx.cmake
@@ -8,7 +8,7 @@ if (NOT NGINX_VERSION)

   string(REGEX MATCH "[0-9]+\\.\[0-9]+\\.[0-9]+" NGINX_VER ${NGINX_VERSION_STRING})
 else()
-  set(NGINX_VER "1.18.0")
+  set(NGINX_VER ${NGINX_VERSION})
 endif()

 set(NGINX_VERSION ${NGINX_VER} CACHE STRING "Nginx version to compile against")
@@ -24,10 +24,10 @@ else()
 endif()

 ExternalProject_Add(project_nginx
-  URL "http://nginx.org/download/nginx-${NGINX_VERSION}.tar.gz"
+  URL "file:///tmp/nginx.tar.gz"
   PREFIX "nginx"
   BUILD_IN_SOURCE 1
-  CONFIGURE_COMMAND ./configure ${NGINX_CONFIGURE_ARGS}
+  CONFIGURE_COMMAND ./configure ${NGINX_CONFIGURE_ARGS} --without-http_gzip_module --without-http_rewrite_module --with-threads --user=nginx --group=nginx --with-file-aio
   BUILD_COMMAND ""
   INSTALL_COMMAND ""
 )
